src/main/java/analysis/GoblintAnalysis.java

```diff
@@ -0,0 +1,189 @@
+package analysis;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.util.*;
+import java.util.concurrent.TimeoutException;
+
+import com.ibm.wala.classLoader.Module;
+
+import magpiebridge.core.AnalysisConsumer;
+import magpiebridge.core.ServerAnalysis;
+import magpiebridge.core.MagpieServer;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonIOException;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+
+import org.eclipse.lsp4j.MessageParams;
+import org.eclipse.lsp4j.MessageType;
+
+import org.zeroturnaround.exec.InvalidExitValueException;
+import org.zeroturnaround.exec.ProcessExecutor;
+import org.zeroturnaround.exec.ProcessResult;
+
+import goblintserver.GoblintClient;
+import goblintserver.GoblintServer;
+import goblintserver.Request;
+
+
+public class GoblintAnalysis implements ServerAnalysis {
+
+    private final MagpieServer magpieServer;
+    private final GoblintServer goblintServer;
+    private final GoblintClient goblintClient;
+
+    private File jsonResult = new File("analysisResults.json");
+
+    // private List<String> projectFiles; // for future use
+
+    private final Logger log = LogManager.getLogger(GoblintAnalysis.class);
+
+
+    public GoblintAnalysis(MagpieServer magpieServer, GoblintServer goblintServer, GoblintClient goblintClient) {
+        this.magpieServer = magpieServer;
+        this.goblintServer = goblintServer;
+        this.goblintClient = goblintClient;
+    }
+
+
+    /**
+     * The source of this analysis, usually the name of the analysis.
+     *
+     * @return the string
+     */
+
+    public String source() {
+        return "GobPie";
+    }
+
+
+    /**
+     * The method that is triggered to start a new analysis.
+     *
+     * @param files    the files that have been opened in the editor (not using due to using the compilation database).
+     * @param consumer the server which consumes the analysis results.
+     * @param rerun    tells if the analysis should be reran.
+     */
+
+    @Override
+    public void analyze(Collection<? extends Module> files, AnalysisConsumer consumer, boolean rerun) {
+        if (rerun) {
+            if (consumer instanceof MagpieServer) {
+
+                preAnalyse();
+
+                MagpieServer server = (MagpieServer) consumer;
+                if (reanalyse()) server.consume(new ArrayList<>(readResultsFromJson()), source());
```

Comment:
The results JSON is read immediately after sending the analyze request? That doesn't give any time for Goblint to analyze.
The sensible thing would be to wait for the analyze request JSON-RPC response to know when it's done and only then read the results. Or even use the `messages` request to get the messages JSON over JSON-RPC instead of going through the file.